lib/rust/ensogl/component/drop-down/src/lib.rs

```diff
@@ -0,0 +1,272 @@
+        /// requested again.
+        ///
+        /// Entries that are provided past currently set `set_number_of_entries` limit will be
+        /// discarded. Make sure to set the correct number of entries before providing entries.
+        provide_entries_at_range(Range<usize>, Vec<T>),
+
+        /// Set the maximum number of entries that can be kept in memory at once. If entry buffer
+        /// exceeds this size, some unused entries will be unloaded and can be requested again. The
+        /// pattern in which entries are unloaded is not defined and should not be relied upon.
+        set_max_cached_entries(usize),
+
+        /// Move the focus to the entry above current focus. If there is no entry focused,
+        /// the first currently visible entry will be focused.
+        focus_previous_entry(),
+        /// Move the focus to the entry below current focus. If there is no entry focused,
+        /// the last currently visible entry will be focused.
+        focus_next_entry(),
+        /// Toggle currently focused entry. If only one entry can be selected, this will deselect
+        /// other entry.
+        toggle_focused_entry(),
+    }
+    Output {
+        /// Emitted when the dropdown needs a list of entries in a specified range to be loaded.
+        /// Will be emitted for the first time after `set_number_of_entries` is set to non-zero
+        /// value. The side handling this event should provide the requested entries using
+        /// `provide_entries_at_range`. Until the requested entries are provided, a loader
+        /// placeholder will be displayed when those entries are in view.
+        entries_in_range_needed(Range<usize>),
+        /// The range of entry indices that are currently in view. This range is different from the
+        /// range of entries that are currently loaded or requested to be loaded.
+        currently_visible_range(Range<usize>),
+        /// Currently selected entries. Changes each time the user selects or deselects an entry.
+        selected_entries(HashSet<T>),
+        /// Currently selected single entry. Is `None` when more than one entry is selected. When
+        /// working with multiselect dropdown, use `selected_entries` instead.
+        single_selected_entry(Option<T>),
+
+        /// Whether or not the dropdown is currently open.
+        is_open(bool),
+    }
+}
+
+
+impl<T: DropdownValue> component::Frp<Model<T>> for Frp<T> {
+    fn init_inputs(frp: &Self::Public) {
+        frp.set_max_size(DEFAULT_SIZE);
+        frp.set_color(DEFAULT_COLOR);
+        frp.set_multiselect(false);
+        frp.set_max_cached_entries(DEFAULT_MAX_ENTRIES);
+        frp.set_open(false);
+        frp.allow_deselect_all(false);
+    }
+
+    fn init(
+        network: &frp::Network,
+        api: &Self::Private,
+        _app: &Application,
+        model: &Model<T>,
+        _style: &StyleWatchFrp,
```

Comment:
why do you keep unused vars here?

Reply:
This is a trait impl. I can't just remove the arguments.